src/runtime_src/core/tools/common/SmiWatchMode.cpp

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2025 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "SmiWatchMode.h"
+#include "core/common/time.h"
+
+// 3rd Party Library - Include Files
+#include <algorithm>
+#include <atomic>
+#include <chrono>
+#include <csignal>
+#include <iomanip>
+#include <sstream>
+#include <thread>
+
+// ------ S T A T I C   V A R I A B L E S -------------------------------------
+namespace signal_handler {
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables) - Required for signal handling
+  std::atomic<bool> g_watch_interrupted{false};
+
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables) - Required for signal handling
+  void (*g_old_signal_handler)(int) = nullptr;
+
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables) - Required for signal handling
+  bool g_signal_handler_set = false;
+
+  void watch_signal_handler(int signal) {
```

Comment:
When namespaced, you should use much simpler names, like just `watch`, it will be called as `signal_handler::watch(..)`